### ISSUE TITLE
Making login + solid communication work

### ISSUE THREAD

**Issue by FdezAriasSara** [Arquisoft/lomap_en3b#41]
trying to solve some bugs related to our first version of the login and pod handling
- [x] handlingl asyncronous communication (either with awaits or handling returned promises)
- [ ]  stay logged in after refreshing
- [x] Lomap folder is created after login if it does not exist, and found if it does.

**Comment by FdezAriasSara** [comment]
Im working on solving some errors of the first version of the implementation. I think it was not working due to  asyncronous messages not being processed properly. Some promises were being ignored F.e:

![image](https://user-images.githubusercontent.com/79209608/226122949-2e22f2c2-bb62-47e5-8112-dc834a0bfa8b.png)

**Comment by FdezAriasSara** [comment]
![image](https://user-images.githubusercontent.com/79209608/226123003-0284cedc-deb3-47a1-a7e6-b196cd5ac451.png)
for example, here i think the await was missing too
![image](https://user-images.githubusercontent.com/79209608/226123013-56e9b601-fdb9-4e33-bcdc-003eade90045.png)

**Comment by FdezAriasSara** [comment]
After adding some awaits, it seems to be working :) 
![image](https://user-images.githubusercontent.com/79209608/226123306-6309599a-533c-4f5f-a1d0-39b7794daf81.png)

**Comment by FdezAriasSara** [comment]
Ok , so after that, a new error arised:


I did this as a patch , i will try to refactor the code into a better solution
![image](https://user-images.githubusercontent.com/79209608/226126846-e149eaf5-320d-4139-bbf6-c45fa5a52b68.png)


Now the problem is related to permissions only, and I will try to fix that too.

![image](https://user-images.githubusercontent.com/79209608/226127049-49eef0cc-f9e3-411a-87cd-6e8d55c71a4e.png)

**Comment by FdezAriasSara** [comment]
@Manuhcuartas  , in podHandler.js

```
export async function checkForLomap(webid) {
    let anyContainer = false;
    let pods = await getPodUrlAll(webid, {fetch : fetch});
    let podWithFolder;
    alert("Your webId: "+webid);
    let i = 0;
    while (!anyContainer && i < pods.length) {
        anyContainer = await checkForLomapInPod(pods[i]);
        if (anyContainer) {
            podWithFolder = pods[i];
        }
        if (i === pods.length - 1) {
          await  createLomapContainer(pods[i]);
        }
        i++;
        
  
}
}
```

what is podWithFolder  meant to do ? Because it has a local scope and a value is assigned if the container is found, but i dont understand how it's used? I mean : 
i understand that it is something like:"While there are pods left , and none of them has been found to have a lomap folder, we keep checkin  for that folder in all pods associated to the given webid, For each one of them: 
if it has a lomap folder we asign the podWithFolder variable with the pod itself.
and in case no pod was found to have a folder, we create the folder in the last one.?

Still , is that podwithfolder being used somewhere else? Or is it never used?

I was thinking to try and implement the following "flow"
![image](https://user-images.githubusercontent.com/79209608/226132854-146c4f1b-8d2a-462e-84d1-f0067f933b9b.png)
so the user has to actually consent to creating the folder else, we will force logout.

**Comment by Manuhcuartas** [comment]
> @Manuhcuartas , in podHandler.js
> 
> ```
> export async function checkForLomap(webid) {
>     let anyContainer = false;
>     let pods = await getPodUrlAll(webid, {fetch : fetch});
>     let podWithFolder;
>     alert("Your webId: "+webid);
>     let i = 0;
>     while (!anyContainer && i < pods.length) {
>         anyContainer = await checkForLomapInPod(pods[i]);
>         if (anyContainer) {
>             podWithFolder = pods[i];
>         }
>         if (i === pods.length - 1) {
>           await  createLomapContainer(pods[i]);
>         }
>         i++;
>         
>   
> }
> }
> ```
> 
> what is podWithFolder meant to do ? Because it has a local scope and a value is assigned if the container is found, but i dont understand how it's used? I mean : i understand that it is something like:"While there are pods left , and none of them has been found to have a lomap folder, we keep checkin for that folder in all pods associated to the given webid, For each one of them: if it has a lomap folder we asign the podWithFolder variable with the pod itself. and in case no pod was found to have a folder, we create the folder in the last one.?
> 
> Still , is that podwithfolder being used somewhere else? Or is it never used?
> 
> I was thinking to try and implement the following "flow" ![image](https://user-images.githubusercontent.com/79209608/226132854-146c4f1b-8d2a-462e-84d1-f0067f933b9b.png) so the user has to actually consent to creating the folder else, we will force logout.

PodWithFolder is supposed to be a local variable that stores the pod that has the lomap folder, (in case there is one).

**Comment by FdezAriasSara** [comment]
But what is it used for? @Manuhcuartas

**Comment by Manuhcuartas** [comment]
> But what is it used for? @Manuhcuartas

For nothing at the moment, but I thought it would be needed in order to store or retrieve the information. The while loop is iterating through all the pods of a webId looking for one that has the lomap folder in it. In case such pod is found, it is stored in the variable PodWithFolder, and my idea is that this variable was passed as a parameter to the proper functions (i.e. GetLocationFromPod(...) or AddLocationToPod(...) as it stores the pod in which the application data is stored for that webId).

**Comment by Manuhcuartas** [comment]
@FdezAriasSara However now that I am reviewing the code, we could get rid of such variable

**Comment by FdezAriasSara** [comment]
> @FdezAriasSara However now that I am reviewing the code, we could get rid of such variable

Actually , I've been working on this in order to make the flow explained in  one of the previous responses. Im trying to manage  permissions and that variable ends up being useful. Right now im at this point: 

![image](https://user-images.githubusercontent.com/79209608/226199188-dbe15fba-50ab-46d5-bba1-5464b698f36d.png)
but still, it does not allow me to make that fetch, resulting in the following error. 

![image](https://user-images.githubusercontent.com/79209608/226199160-b8b3fe48-9167-4859-b9cd-168f99cc7bc7.png)
 
Ill keep trying to solve this

**Comment by FdezAriasSara** [comment]
After some trial and error , i realized that we need to be granted read and write access before even trying to search for the lomap folder..
work so far:

https://github.com/Arquisoft/lomap_en3b/commit/19dde54ca794e35476dd9ac68f50354d5f703216#commitcomment-105469083

**Comment by FdezAriasSara** [comment]
Finally i got lomap to be able to find the dataset of a non public profile resource:
-I added handleIncomingRedirect to the <LoginButton/> component of solid's react sdk
that way, when user is redirected the session will be restored.
-I also made pod.replace("/profile/card#me","/")-> because getPodsUrlAll() was returning the profile url, and i needed the base pod url
-I also ended up doing something and not too happy with, but that works. I pass the session object so i can pass  in the options array {fetch: session.fetch } for methods like getSolidDataset() . That way the request is authenticated, and so i do retrieve the dataset.
fore example inside : checkForLomap(session) 

    ` let pods = await getPodUrlAll(session.info.webId, {fetch : session.fetch});.`

its weird because i get 
![image](https://user-images.githubusercontent.com/79209608/227054002-f2237a94-5112-43c6-a7e3-4f515f48e125.png)

Is like theres a span where the requests are not authorized, i dont know why? I will dig into it.

ANYWAYS: if no folder the folder is actually created ! :) 

NOW i will try to focus back on rendering location pins, since at least i can get information from the pod.
